File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/api-response-models/response-course-model.js

import ViewCourse from "../views/view-course";
import ResponseCommentModel from "./response-comment-model";

class ResponseCourseModel extends ViewCourse {
    constructor() {
        super();
        this.NumberComments = null;
        this.TopComments = null;
        this.IsMarked = null;
        this.AverageStar = null;
        this.MyStars = null;
        this.TotalStar = null;
        this.Role = null;
        this.Categories = [];
        this.CourseRoleType = null;
        this.CourseRelationId = null;
    }

    init() {
        return new ResponseCourseModel();
    }

    copy(entity) {
        super.copy(entity);
        if (this.TopComments?.length > 0){
            this.TopComments = this.TopComments.map(function(com){
                let comm = new ResponseCommentModel();
                comm.copy(com);
                return comm;
            });
        }
        return this;
    }
}

export default ResponseCourseModel;

File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/ApiResponseModels/ResponseCourseModel.cs
using KnowledgeSharingApi.Domains.Enums;
using KnowledgeSharingApi.Domains.Interfaces.ModelInterfaces.ApiResponseModelInterfaces;
using KnowledgeSharingApi.Domains.Models.Entities.Tables;
using KnowledgeSharingApi.Domains.Models.Entities.Views;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.ApiResponseModels
{
    public class ResponseCourseModel : ViewCourse, IResponseCourseModel
    {
        public int NumberComments { get; set; }
        public List<ResponseCommentModel> TopComments { get; set; } = [];
        public bool IsMarked { get; set; }
        public double? AverageStars { get; set; }
        public double? MyStars { get; set; }
        public int TotalStars { get; set; }
        public ECourseRoleType? Role { get; set; }
        public List<string> Categories { get; set; } = [];
        public ECourseRoleType? CourseRoleType { get; set; }
        public Guid? CourseRelationId { get; set; }
    }
}
